### PR TITLE
fix: cypress test added for forking critical issue

### LINKED ISSUE #24702: [Bug]: On forking an app with the mongo we get an error, and are unable to fork
On forking an app with the above connection string URI set to Yes, we get an error, and are unable to fork with this setting. [Workaround, with the use connection string set to no, and all details entered, we can proceed with forking]
Screenshot 2023-06-20 at 1 52 57 PM

_Originally posted by @btsgh in https://github.com/appsmithorg/appsmith/issues/24660#issuecomment-1598338428_

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR adds cypress test to ensure the fix made for issue #24702 
#### PR fixes following issue(s)
Fixes #24702 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by sneha122** [comment]
Testing will not be required for this PR as it simply adds cypress test for forking functionality @prapullac @btsgh

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      cy.get("@guid").then((uid) => {
+        forkWorkspaceName = "ForkApplication" + uid;
+        homePage.CreateNewWorkspace(forkWorkspaceName);
+        homePage.FilterApplication(currentAppName);
+
+        agHelper.Sleep(500);
+
+        homePage.ForkApplication(currentAppName, forkWorkspaceName);
+
+        dataSources.FillMongoDSForm();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: The bug #24702 is for mongo URI connection error. Here I see we are filling it normally. Can we add test case for bug specific scenario please?

Comment (sneha122) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: @Aishwarya-U-R Do we have existing utility functions for connecting to mongo using URI?

Comment (sneha122) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: @Aishwarya-U-R Thanks will update accordingly

Comment (sneha122) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: @Aishwarya-U-R I have seen the uri being there in hostports.ts file, I hope using that will give the required results

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: No, not sure if that is still working. Better to get the TED MONGO URI formation from @btsgh & use it in test.

Comment (btsgh) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: Girls - am not able to find the URI string .. need some more time on this.. will buzz as soon as I can on this.

Comment (btsgh) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: Found the string that works - 
mongodb://0.tcp.in.ngrok.io:11330/mongo_samples

Please replace the ngrok ip and port accordingly. @Aishwarya-U-R @sneha122

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: Wonderful, so this is `mongodb://host.docker.internal:28017/mongo_samples` for TED. @sneha122 - please use this & re-write the test.

Comment (sneha122) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: @Aishwarya-U-R I replaced host.docker.internal with localhost, and tried this url mongodb://localhost:28017/mongo_samples -> says test invalid. The link which @btsgh gave would only work in local right? How would it work on CI?

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:38]: What I have should work for Ci. Did that not work?

File: app/client/cypress/support/Objects/Hostport.ts [hunk 2]
```diff
@@ -3,8 +3,7 @@ export class DefaultHostPort {
   mongo_host = "host.docker.internal";
   mongo_port = 28017;
   mongo_databaseName = "mongo_samples";
-  mongo_uri =
-    "mongodb+srv://localhost:localhost@mockdb.net/mock?retryWrites=true&w=majority";
+  mongo_uri = "mongodb://host.docker.internal:28017/mongo_samples";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/Hostport.ts:6]: Can this please be -     `mongo_uri = `mongodb://${this.mongo_host}:${this.mongo_port}/${this.mongo_databaseName}`;
`

**Comment by Aishwarya-U-R** [comment]
> /ok-to-test

@sneha122 - Always merge latest manually & then start oktotest. Have seen the auto merge with release not really consistent every time.

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import {
+  dataSources,
+  homePage,
+  agHelper,
+} from "../../../../support/Objects/ObjectsCore";
+
+let currentWorkspace: string, currentAppName: string, forkWorkspaceName: string;
+
+describe("Fork application across workspaces", function () {
+  it("1. Signed user should be able to fork a public forkable app & Check if the forked application has the same dsl as the original", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:10]: Can we add Bug id to `it` description?

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    homePage.NavigateToHome();
+    // Create new workspace to create App in
+    agHelper.GenerateUUID();
+    cy.get("@guid").then((uid) => {
+      currentWorkspace = "CurrentWorkspace " + uid;
+      currentAppName = "MongoQueryApp " + uid;
+      homePage.CreateNewWorkspace(currentWorkspace);
+      homePage.CreateAppInWorkspace(currentWorkspace, currentAppName);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:18]: Everytime a spec starts, we are already in a new application right? Do we still need this step?

Comment (sneha122) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:18]: @Aishwarya-U-R Yes I agree, but I was not able to get the workspace name for the current workspace, hence decided to create a different workspace altogether for this test, later on in the test, I am also deleting both applications and workspaces as well

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ForkApplicationReconnectModal_spec.ts:18]: Ok got it, there is no issue in creating it, but I was thinking why to create & why to delete it & why spend extra time in such activities. And I get your point of isolation.
This is okay, but lets use existing new app created already for the new testcase too, hereon!

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress changes!

**Comment by Aishwarya-U-R** [comment]
@sneha122 - Will add TestPlanApproved label - after the case is a pass in CI run!

**Pull request merged**